### PR TITLE
Reimplement ManufacturerInfo with multiplexer transformation

### PR DESCRIPTION
**Pull request by mmandlis**
Misc fixes:
- Product multiplexer: set positive rank
- Description
  . Maintain the correct order (by rank) of particles combined into a suggestion 
  . Capitalize first letter of resolved pattern's token in dom formatter 
- Hosted slots: distinguish xen templates of the same transformation particle but different hosted particles.

PS: I still keep the original particles for now, because they are helpful in debugging.

https://github.com/PolymerLabs/arcs/issues/299

### CODE REVIEW

File: runtime/browser/demo/recipes.manifest [hunk 1]
```diff
@@ -56,18 +57,10 @@ recipe
...
-recipe
-  map as view1
-  use as view2
-  use as view3
-  GiftList
-    person <- view1
+#  Arrivinator
+#    list <- view2
+#  Arrivinator
+#    list <- view3
```
Comment (shans) [inline runtime/browser/demo/recipes.manifest]: delete the commented out lines.

File: runtime/browser/demo/recipes.manifest [hunk 2]
```diff
@@ -78,8 +71,11 @@ recipe
 # Check manufacturer information for products.
 recipe
   use #shortlist as shortlist
-  ManufacturerInfo
-    list <- shortlist
+#  ManufacturerInfo
+#   list <- shortlist
```
Comment (shans) [inline runtime/browser/demo/recipes.manifest]: & here

File: runtime/description-dom-formatter.js [hunk 3]
```diff
@@ -83,7 +92,16 @@ export default class DescriptionDomFormatter extends DescriptionFormatter {
         template = template.concat(`${index == 0 && i == 0 ? token.text[0].toUpperCase() + token.text.slice(1) : token.text}`);
       } else {  // view or slot handle.
         let sanitizedFullName = token.fullName.replace(/[.{}_\$]/g, '');
-        template = template.concat(`<span>{{${sanitizedFullName}}}</span>`);
+        let attribute = '';
+        if (i == 0) {
+          // Capitalize the first letter in the token.
+          template = template.concat(`<style>
+            [firstletter]::first-letter { text-transform: capitalize; }
```
Comment (shans) [inline runtime/description-dom-formatter.js]: nit: this is actually a lot more expensive than just capitalizing the data directly in the model.

File: runtime/browser/demo/recipes.manifest (hunk 1 above)
Comment (mmandlis) [reply runtime/browser/demo/recipes.manifest]: done

File: runtime/browser/demo/recipes.manifest (hunk 2 above)
Comment (mmandlis) [reply runtime/browser/demo/recipes.manifest]: done

File: runtime/description-dom-formatter.js (hunk 3 above)
Comment (mmandlis) [reply runtime/description-dom-formatter.js]: added a TODO - i don't resolve the data in the model until later, and right now i don't have a way to remember to capitalize it then.

**Pull request merged**